### DISCUSSION TITLE
Certificate Template - QR code size

### DISCUSSION THREAD

**Discussion by maheshkumargangula** [Sunbird-RC/community#214] [Issues]
Hi @dileepbapat,

We observed that a small QR code (~0.7inch) in the certificate not working properly to scan and verify. So, can you please let us know the recommended QR image size (width and height in the certificate template) for a certificate generated using sunbird RC?

Cc: @vrayulu

**Comment by pramodkvarma** [comment]
Are we using the URL to QR code as we discussed (for backward compatibility)? If so the URL itself is not the VC format, rather once the URL is scanned, it takes user to a page where VC details are displayed, right? Or are we having both offline and URL versions? 

@dileepbapat @tejash-jl can you please urgently help?

**Reply by di1** [reply]
Yes, 1. offline, 2. URL with payload and now 3. short URL being added with https://github.com/Sunbird-RC/sunbird-rc-core/pull/112

**Reply by pramodkvarma** [reply]
Sorry, I am confused and didn't understand your response.

**Reply by dileepbapat** [reply]
QR code type is configurable, allows 3 mechanism
1. Offline format where complete w3c json payload is embedded in QR code. However on scanning generic scanner will not be able to interpret it
2. URL with W3C signed json as query param. Can be usable with generic scanner to navigate to online version of certificate. Custom verifier will be able to read the query param and validate (even in offline). This option comes with URL **query parameter size** as limitation
3. Only URL which points to certificate (supports only **online** mode). This was recently added, advantage is QR code payload is smaller, easy to scan and smaller size QR code is enough for print and scan.

**Reply by pramodkvarma** [reply]
Thank you for clarifying @dileepbapat. This is really good that RC is now flexible to accommodate all 3 scenarios.

**Comment by dileepbapat** [comment]
QR Code print size depends on print, paper quality and size of signed json payload. From what we tested, when printed with Inkjet printer on A4 page: 2-3 inch QR code is usable with signed json is about 1kb.

Smaller QR code works if certificate is laser printed on quality paper.
